### ISSUE TITLE
Unable to bootstrap clingo from sources

### ISSUE THREAD

**Issue by behrmann** [LLNL/spack#30937]
### Steps to reproduce

It currently seems impossible to bootstrap clingo from sources without using pre-built binaries, simply by untrusting the pre-built binaries.

```console
$ git clone --config feature.manyFiles=true --branch releases/v0.18 https://github.com/spack/spack.git spack
Cloning into 'spack'...
remote: Enumerating objects: 387505, done.
remote: Counting objects: 100% (3337/3337), done.
remote: Compressing objects: 100% (2873/2873), done.
remote: Total 387505 (delta 478), reused 596 (delta 404), pack-reused 384168
Receiving objects: 100% (387505/387505), 178.64 MiB | 31.16 MiB/s, done.
Resolving deltas: 100% (164270/164270), done. 
$ source spack/share/spack/setup-env.sh 
$ spack bootstrap untrust github-actions-v0.1
==> "github-actions-v0.1" is now untrusted and will not be used for bootstrapping
$ spack bootstrap untrust github-actions-v0.2
==> "github-actions-v0.2" is now untrusted and will not be used for bootstrapping
$ spack solve zlib
==> Bootstrapping clingo from pre-built binaries
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-z6v6zvc6awioeompbvo735b4flr3yuyz.spec.json
==> Fetching https://mirror.spack.io/bootstrap/github-actions/v0.2/build_cache/linux-centos7-x86_64/gcc-10.2.1/clingo-bootstrap-spack/linux-centos7-x86_64-gcc-10.2.1-clingo-bootstrap-spack-z6v6zvc6awioeompbvo735b4flr3yuyz.spack
==> Installing "clingo-bootstrap@spack%gcc@10.2.1~docs~ipo+python+static_libstdcpp build_type=Release arch=linux-centos7-x86_64" from a buildcache
==> Bootstrapping patchelf from pre-built binaries
==> Bootstrapping patchelf from pre-built binaries
==> Bootstrapping patchelf@0.13.1:0.13.99 %gcc target=x86_64 from sources
==> Installing patchelf-0.13.1-thfxinawpnopkdelwjwry42dbt3iwk5p
==> No binary for patchelf-0.13.1-thfxinawpnopkdelwjwry42dbt3iwk5p found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/08/08c0237e89be74d61ddf8f6ff218439cdd62af572d568fb38913b53e222831de.tar.gz
==> No patches needed for patchelf
==> patchelf: Executing phase: 'autoreconf'
==> patchelf: Executing phase: 'configure'
==> patchelf: Executing phase: 'build'
==> patchelf: Executing phase: 'install'
==> patchelf: Successfully installed patchelf-0.13.1-thfxinawpnopkdelwjwry42dbt3iwk5p
  Fetch: 0.12s.  Build: 6.27s.  Total: 6.39s.
[+] /home/user/.spack/bootstrap/store/linux-debian11-x86_64/gcc-10.2.1/patchelf-0.13.1-thfxinawpnopkdelwjwry42dbt3iwk5p
==> Best of 1 considered solutions.
==> Optimization Criteria:
  Priority  Criterion                                            Installed  ToBuild
  1         number of input specs not concretized                        -        0
  2         number of packages to build (vs. reuse)                      -        0
  3         deprecated versions used                                     0        0
  4         version weight                                               0        0
  5         number of non-default variants (roots)                       0        0
  6         preferred providers for roots                                0        0
  7         default values of variants not being used (roots)            0        0
  8         number of non-default variants (non-roots)                   0        0
  9         preferred providers (non-roots)                              0        0
  10        compiler mismatches                                          0        0
  11        OS mismatches                                                0        0
  12        non-preferred OS's                                           0        0
  13        version badness                                              0        0
  14        default values of variants not being used (non-roots)        0        0
  15        non-preferred compilers                                      0        0
  16        target mismatches                                            0        0
  17        non-preferred targets                                       30        0

zlib@1.2.12%gcc@10.2.1+optimize+pic+shared patches=0d38234 arch=linux-debian11-skylake
```

This can be worked around by making a local mirror and trusting that one
```console
$ spack bootstrap mirror $PWD/bootstrap_mirror/
$ spack bootstrap add --trust local-sources <final-path>/metadata/sources
```
but this seems not intended.

As an aside: The documentation is also somewhat wrong, as `spack bootstrap untrust github-actions` no longer works, but needs `-v0.1` and/or `-v0.2`.

### Information on your system

* **Spack:** 0.18.0 (c09bf37ff690c29779a342670cf8a171ad1b9233)
* **Python:** 3.9.2
* **Platform:** linux-debian11-skylake
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have searched the issues of this repo and believe this is not a duplicate
- [X] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
Fixed by #31042 (the fix will be part of the next patch release)

**Comment by behrmann** [comment]
Thanks @alalazo!
